external/model-preparation-algorithm/mpa_tasks/utils/data_utils.py

```diff
@@ -17,7 +17,8 @@ def get_cls_img_indices(labels, dataset):
     for i, item in enumerate(dataset):
         item_labels = item.annotation_scene.get_labels()
         for i_l in item_labels:
-            img_indices[i_l.name].append(i)
+            if i_l in labels:
+                img_indices[i_l.name].append(i)
```

Comment:
Why don't we make img_indices as defaultdict(list)?
Could you explain why there are some labels in annotations are missing in 'labels'?

Reply:
This error occurred when det -> cls task transition. `get_labels()` brings all labels the item contains no matter what the domain is. The label for the previous det task is brought to item_labels. But the keys(classes) in img_indices has only classification domain(It means label schema is working well). To check if item_label is cls or else, that `if i_l in labels` is needed. If we use defaultdict(list) and let img_indices make keys from item_labels without checking the domain, we cannot extract the labels of the classification domain.